### PR TITLE
organize imports

### PR DESCRIPTION
**Pull request by anoblet**
This adds `prettier-plugin-organize-imports` (https://github.com/simonhaenisch/prettier-plugin-organize-imports) which handles sorting of imports automatically.

### CODE REVIEW

**Comment by anoblet** [comment]
I'm not sure if prettier should be manually run on `cxl-lumo-styles`, `cxl-ui`, and `storybook` separately to only pick up the import organization or if it's better to run on `packages` in it's entirety to get everything up to speed.

**Review by pawelkmpt — APPROVED** [review]

File: packages/cxl-ui/src/index.js [hunk 1]
```diff
@@ -2,21 +2,19 @@ import '@vaadin/vaadin-notification';
...
+export { CXLPaywallElement } from './components/cxl-paywall.js';
+export { CXLPlaybookAccordionElement } from './components/cxl-playbook-accordion.js';
+export { CXLPlaybookProgressBarElement } from './components/cxl-playbook-progress-bar.js';
+export { CXLSaveFavoriteElement } from './components/cxl-save-favorite.js';
 export { CXLSectionElement } from './components/cxl-section.js';
+export { CXLStarRatingElement } from './components/cxl-star-rating.js';
 export { CXLTabsSliderElement } from './components/cxl-tabs-slider.js';
-
 // Order matters.
 export { CXLVaadinAccordionElement } from './components/cxl-vaadin-accordion.js';
```
Comment (lkraav) [inline packages/cxl-ui/src/index.js:18]: Are we sure we're not creating a problem here?

I remember this import order possibly affects how style modules are loaded, which may or may not be a thing for newest vaadin-themeable-mixin, but that's yet to be merged.

File: package.json [hunk 2]
```diff
@@ -76,6 +77,7 @@
     "stylelint": "^13.6.1",
     "stylelint-config-prettier": "^8.0.0",
     "stylelint-config-twbs-bootstrap": "^2.1.0",
+    "typescript": "^4.5.5",
```
Comment (lkraav) [inline package.json:80]: Typescript is a requirement for this, or...?

File: package.json [hunk 3]
```diff
@@ -67,6 +67,7 @@
     "npm-run-all": "^4.1.5",
     "polymer-webpack-loader": "^2.0.3",
     "prettier": "^2.2.1",
+    "prettier-plugin-organize-imports": "^2.3.4",
```
Comment (lkraav) [inline package.json:70]: Wasn't there an ESLint rule to do this that Vaadin used https://github.com/vaadin/web-components/pull/2915

Why wouldn't we stay as close as possible to that?

File: packages/cxl-ui/src/index.js (hunk 1 above)
Comment (anoblet) [reply packages/cxl-ui/src/index.js:18]: I'll see if there's a way for us to omit certain files.

File: package.json (hunk 2 above)
Comment (anoblet) [reply package.json:80]: Typescript is a dependency for `prettier-plugin-organize-imports`.

File: package.json (hunk 3 above)
Comment (anoblet) [reply package.json]: I'll switch to `eslint-plugin-simple-import-sort`.

**Comment by anoblet** [comment]
I switched to `eslint-plugin-simple-import-sort`, omitted the code block in `packages/cxl-ui/index.js` from being sorted and ran `eslint` only on `cxl-ui` and `storybook` for a clean PR.

**Comment by anoblet** [comment]
Updated for the `playbooks` base branch.

**Comment by anoblet** [comment]
I could get rid of the actual eslint run in this PR and open up separate PRs for master and playbooks that run eslint on the entire packages folder?

**Comment by lkraav** [comment]
> I could get rid of the actual eslint run in this PR and open up separate PRs for master and playbooks that run eslint on the entire packages folder?

It already runs on commit

https://github.com/conversionxl/aybolit/blob/master/package.json#L18

I thought everything ESLinted by now, but if it does find differences, yes, we could PR that.

**Comment by anoblet** [comment]
Running eslint on the packages folder triggers prettier to clean up some left over stuff. I created a separate PR for master: https://github.com/conversionxl/aybolit/pull/192

**Comment by pawelkmpt** [comment]
Not clear if useful as not merged for 1,5 year.

**Comment by anoblet** [comment]
This would be nice to have at some point as it standardizes the order in which we define imports. I have a habit of pressing `Alt + Shift + O` so that imports are sorted alphabetically.

**Pull request closed without merging**